zookeeper-server/src/main/java/org/apache/zookeeper/server/NIOServerCnxnFactory.java

```diff
@@ -178,6 +178,14 @@ public void run() {
             try {
                 while (!stopped && !acceptSocket.socket().isClosed()) {
                     try {
+                        /** if zkServer is null, don't accept the connection, until not null.
+                         *  if accept the connection, when reading data from socket, it will be closed by null of zkServer
+                         *
+                         */
+                        if (zkServer == null) {
```

Comment:
shall we sleep a little time ?
otherwise we are in a busy loop

Reply:
@eolivelli 
No need. The loop is ok.  If sleep a little time, maybe cause the clients waiting to connect to servers.